web/pgadmin/__init__.py

```diff
@@ -915,13 +917,15 @@ def all_exception_handler(e):
     @app.errorhandler(HTTPException)
     def http_exception_handler(e):
         current_app.logger.error(e, exc_info=True)
-        if e.code == 400 and\
-                e.description == 'The CSRF session token is missing.':
-            error = str(e.description) + 'Please refresh the page.'
-            return internal_server_error(errormsg=gettext(error))
         return e
 
-    # Intialize the key manager
+    # Send unauthorized response if CSRF errors occurs.
+    @app.errorhandler(CSRFError)
+    def handle_csrf_error(error):
+        err_msg = str(error.description) + ' You need to refresh the page.'
+        return unauthorized(errormsg=gettext(err_msg))
```

Comment:
gettext should be applied to the string, not to the variable.

Reply:
Done